File: Introduction_to_Algorithms/anagram_checker.cpp
# include <bits/stdc++.h>
using namespace std;
typedef map<char,int> dictionary;

auto is_anagram(string a,string b)
{
 if(a.length() != b.length())
 {
  return false;
 }
 dictionary freq_a;
 for(auto i : a)
     freq_a[i]++;
dictionary freq_b;
  for(auto i : b)
     freq_b[i]++;
if(equal(begin(freq_a),end(freq_a),begin(freq_b)))
   return true;
return false;
}
int main()
{
cout<<(is_anagram("eat","toa")? "Yes":"No");
}

File: Introduction_to_Algorithms/anagram_checker.rb
require 'benchmark'
def anangram?(a,b)
	return false if a.length != b.length
	 freq_a = Hash.new 0
	 freq_b = Hash.new 0
	 a.chars { |ch| freq_a[ch] += 1}
	 b.chars { |ch| freq_b[ch] += 1}
	 if freq_a == freq_b
		 return true
          else
	      return false
	 end
	
end


Benchmark.bm do |bm|
	bm.report { puts anangram?("tea","eat") ? " Yes" : "No" }
end